openassessment/workflow/models.py

```diff
@@ -353,11 +422,27 @@ def set_score(self, score):
                 'points_possible'.
 
         """
-        sub_api.set_score(
-            self.submission_uuid,
-            score["points_earned"],
-            score["points_possible"]
-        )
+        if not self.staff_score_exists():
```

Comment:
Why does this check have to be done? Won't the workflow ensure that the staff score will beat out any other assigned scores?